### DISCUSSION TITLE
Does OpenSTA have any plans to support power analysis by reading SAIF files?

### DISCUSSION THREAD

**Discussion by Zhihui-SJTU** [parallaxsw/OpenSTA#97] [Q&A] [answered]
I noticed there is an example in the examples/ directory that demonstrates reading a VCD simulation file. I'm wondering if there are any plans to support reading SAIF files, and if so, when it might be available?

**Comment by jjcherry56** [comment, marked as answer]
It makes sense to support it. It is just another thing on an infinite list. I can't agree that it is easier to generate saif than vcd, since iverilog can produce vcd but not saif. I have a reader close to working I should be able to push later this week.

**Reply by Zhihui-SJTU** [reply]
I'm glad to hear that you have a nearly finished SAIF reader. I'm really looking forward to trying it out!

**Reply by Zhihui-SJTU** [reply]
I noticed that in the header of the provided gcd_sky130hd.saif file, there is a line (PROGRAM_NAME "vcd2saif.lisp"). Does this tool exist in OpenSTA?

**Comment by jjcherry56** [comment]
No. I tried using https://github.com/ZHANG-Qidi/VCD2SAIF/tree/main but it has a number of bugs so I gave up on it.

**Comment by akashlevy** [comment]
SAIF is fully supported now btw
